impl/client/src/main/javascript/web/util/_a11y.js

```diff
@@ -35,6 +35,358 @@ define(["./_focus"], function(focusUtil) {
+    var clearKeysSoFarHandle = null;
+    var handleFocusChange = function () {};
+
+    registerEvents();
+
+    /**
+     *  Register events for the listbox interactions
+     */
+    function registerEvents() {
+      elem.addEventListener("keydown", onKeyPress);
+      elem.addEventListener("click", onClick);
+    }
+
+    function getActiveOption() {
+      return document.getElementById(activeDescendantId);
+    }
+
+    /**
+     * Check if an item is clicked on. If so, focus on it and select it.
+     *
+     * @param evt - The click event object
+     */
+    function onClick(evt) {
+      if (evt.target.getAttribute("role") !== "option") {
+        return;
+      }
+
+      setActiveOption(evt.target);
+      toggleSelectItem(evt.target);
+      updateScroll();
+
+      if (evt.shiftKey) {
+        selectRange(startRangeIndex, evt.target);
+      }
+    }
+
+    /**
+     * Handle various keyboard controls; UP/DOWN will shift focus; SPACE selects an item.
+     *
+     * @param evt - The keydown event object
+     */
+    function onKeyPress(evt) {
+      var key = evt.which || evt.keyCode;
+      var lastActiveId = activeDescendantId;
+      activeDescendantId = elem.getAttribute("aria-activedescendant");
+      var allOptions = getAllOptions();
+      var currentItem = getActiveOption() || allOptions[0];
+      var nextItem = currentItem;
+
+      if (!currentItem) {
+        return;
+      }
+
+      switch (key) {
+        case keyCodes.pageUp:
+        case keyCodes.pageDown:
+          break;
+        case keyCodes.arrowUp:
+        case keyCodes.arrowDown:
+          evt.preventDefault();
+          if (!activeDescendantId) {
+            // focus first option if no option was previously focused, and perform no other actions
+            setActiveOption(currentItem);
+            break;
+          }
+
+          if (key === keyCodes.arrowUp) {
+            nextItem = findPreviousOption(currentItem);
+          } else {
+            nextItem = findNextOption(currentItem);
+          }
+
+          if (nextItem && event.shiftKey) {
+            selectRange(startRangeIndex, nextItem);
+          }
+
+          if (nextItem) {
+            setActiveOption(nextItem);
+            evt.preventDefault();
+          }
+
+          break;
+        case keyCodes.home:
+          evt.preventDefault();
+          focusFirstItem();
+
+          if (evt.shiftKey && evt.ctrlKey) {
+            selectRange(startRangeIndex, 0);
+          }
+          break;
+        case keyCodes.end:
+          evt.preventDefault();
+          focusLastItem();
+
+          if (evt.shiftKey && evt.ctrlKey) {
+            selectRange(startRangeIndex, allOptions.length - 1);
+          }
+          break;
+        case keyCodes.shift:
+          startRangeIndex = getOptionIndex(currentItem);
+          break;
+        case keyCodes.space:
+          evt.preventDefault();
+          toggleSelectItem(nextItem);
+          break;
+        case 65:
```

Comment:
Missing a key code constant? Could be called `keyCodes.A`, given it's a single letter.